src/Aspire.Hosting.AWS/IAWSSDKConfig.cs

```diff
@@ -0,0 +1,40 @@
+ // Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Amazon;
+using Amazon.Runtime;
+
+namespace Aspire.Hosting.AWS;
+
+/// <summary>
+/// Configuration used to construct service client from the AWS SDK for .NET.
+/// </summary>
+public interface IAWSSDKConfig
+{
+    /// <summary>
+    /// The AWS credential profile to use for resolving credentials to make AWS service API calls.
+    /// </summary>
+    string? Profile { get; set; }
+
+    /// <summary>
+    /// The AWS region to deploy the CloudFormation Stack.
+    /// </summary>
+    RegionEndpoint? Region { get; set; }
+
+    internal T CreateServiceConfig<T>() where T : ClientConfig, new()
+    {
+        var config = new T();
+
+        if(!string.IsNullOrEmpty(Profile))
```

Comment:
nit: 
```suggestion
        if (!string.IsNullOrEmpty(Profile))
```

Formatting everywhere